api/src/env.ts

```diff
@@ -263,8 +263,8 @@ const typeMap: Record<string, string> = {
 
 let env: Record<string, any> = {
 	...defaults,
-	...process.env,
 	...getEnv(),
+	...process.env,
```

Comment:
This effectively reverts https://github.com/directus/directus/pull/12900 and should be extracted to a separate PR